addons/purchase_stock/models/stock_move.py

```diff
@@ -36,7 +36,7 @@ def _get_price_unit(self):
         order = line.order_id
         received_qty = line.qty_received
         if self.state == 'done':
-            received_qty -= self.product_uom._compute_quantity(self.quantity_done, line.product_uom)
+            received_qty -= self.product_uom._compute_quantity(self.quantity_done, line.product_uom, rounding_method='HALF-UP')
```

Comment:
correcting the quantity done on the stock move is the only mandatory change to fix the bug
I've edited the other lines aswell just to have the reserved quantity on the PO match